examples/options/src/generation.cpp

```diff
@@ -49,18 +49,38 @@ generation::generation() : interface("Particle Generation Options") {
         "gen-eta",
         po::value(&eta_range)->value_name("MIN:MAX")->default_value(eta_range),
         "Range of eta");
+    m_desc.add_options()("gen-theta",
+                         po::value(&theta_range)
+                             ->value_name("MIN:MAX")
+                             ->default_value(theta_range),
+                         "Range of theta in degree");
     m_desc.add_options()("particle-type",
                          po::value<int>(&pdg_number)->default_value(pdg_number),
                          "PDG number for the particle type");
 }
 
-void generation::read(const po::variables_map&) {
+void generation::read(const po::variables_map& vm) {
 
     vertex *= detray::unit<float>::mm;
     vertex_stddev *= detray::unit<float>::mm;
     mom_range *= detray::unit<float>::GeV;
     phi_range *= detray::unit<float>::degree;
-    theta_range = eta_to_theta_range(eta_range);
+
+    // The eta and theta range can not be specified at the same time
+    if (vm.count("gen-eta") && !vm["gen-eta"].defaulted() &&
+        vm.count("gen-theta") && !vm["gen-theta"].defaulted()) {
+        throw std::logic_error(
+            std::string("Conflicting options 'gen-eta' and 'gen-theta'"));
+    } else if (vm.count("gen-eta") && !vm["gen-eta"].defaulted()) {
+        theta_range = eta_to_theta_range(eta_range);
+    } else if (vm.count("gen-theta") && !vm["gen-theta"].defaulted()) {
+        theta_range *= detray::unit<float>::degree;
+        eta_range = theta_to_eta_range(theta_range);
+    } else {
+        // When none of eta and theta is specified use the default eta range
+        theta_range = eta_to_theta_range(eta_range);
+    }
```

Comment:
Do we need this very last part? I thought you chose the default values of the `eta_range` and `theta_range` variables such that they would be equivalent. 🤔

Reply:
Removed